### PR TITLE
Mocha/Karma for unit tests

### PR DESCRIPTION
**Pull request by georgebatalinski**
Add test suite - first test for keyboard navigation and infrastructure for future tests

### CODE REVIEW

File: tests/spec/rz-slider-service-test.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    var event = pressLeftArrow();
+    service.onPointerFocus(element, 'rzSliderModel', event);
+    service.onKeyboardEvent(event);
+    expect(scope.rzSliderModel).to.equal(0);
+    done();
+  });
+
+  function pressLeftArrow() {
+    var evt = document.createEvent('CustomEvent');  // MUST be 'CustomEvent'
+    evt.initCustomEvent('yourCustomEventName', false, false, null);
```
Comment (ValentinH) [inline tests/spec/rz-slider-service-test.js]: Why don't you set 'keydown' here instead of yourCustomEventName?

**Comment by ValentinH** [comment]
Great! Why did you modify the demo page, it should not be modified.

In order to be merged, I need you to rebase your branch on the keyboard-support branch because there are merge conflicts.

**Comment by georgebatalinski** [comment]
done :)

**Comment by ValentinH** [comment]
Perfect! thanks :)

**Pull request merged**